server/src/main/java/org/opensearch/gateway/PrimaryShardBatchAllocator.java

```diff
@@ -60,51 +62,42 @@ protected FetchResult<TransportNodesListGatewayStartedShards.NodeGatewayStartedS
     }
 
     @Override
-    public AllocateUnassignedDecision makeAllocationDecision(ShardRouting unassignedShard, RoutingAllocation allocation, Logger logger) {
```

Comment:
This needs to be overriden here - otherwise it will fallback to single fetch which is not supported 

```
    protected FetchResult<TransportNodesListGatewayStartedShards.NodeGatewayStartedShards> fetchData(
        ShardRouting shard,
        RoutingAllocation allocation
    ) {
        logger.error("fetchData for single shard called via batch allocator, shard id {}", shard.shardId());
        throw new IllegalStateException("PrimaryShardBatchAllocator should only be used for a batch of shards");
    }

```

Reply:
Added the method to batch shard allocators.